autogpt_platform/backend/backend/executor/database.py

```diff
@@ -50,17 +54,11 @@ def send_execution_update(self, execution_result: ExecutionResult):
     def exposed_run_and_wait(
         f: Callable[P, Coroutine[None, None, R]]
     ) -> Callable[Concatenate[object, P], R]:
-        @expose
-        @wraps(f)
-        def wrapper(self, *args: P.args, **kwargs: P.kwargs) -> R:
-            coroutine = f(*args, **kwargs)
-            res = self.run_and_wait(coroutine)
-            return res
-
-        # Register serializers for annotations on bare function
-        register_pydantic_serializers(f)
-
-        return wrapper
+        # TODO:
+        #  This function lies about its return type to make the DynamicClient
+        #  call the function synchronously, fix this when DynamicClient can choose
+        #  to call a function synchronously or asynchronously.
+        return expose(cast(Callable[Concatenate[object, P], R], f))
```

Comment:
You can use the inspect module from Python to check this

Reply:
The function lying on its signature is unfortunately intentional.

To make the fast API call behaves like a function call, we need to make the dynamic client (for now) thinks it's calling a sync function, where in fact the server can now write the endpoints in async mode.